src/python/pants/goal/run_tracker.py

```diff
@@ -390,3 +395,38 @@ def shutdown_worker_pool(self):
     N.B. This exists only for internal use and to afford for fork()-safe operation in pantsd.
     """
     SubprocPool.shutdown(self._aborted)
+
+  def report_target_info(self, scope, target, key, val):
+    """
+    Add target information to run_info under target_data
+    :param string scope: The running scope
+    :param string target: The target that we want to store info for
+    :param string or list of items to make into a key key: The key for the info being stored
+    :param dict or string val: The value of the info being stored
+    """
+    # Stringify list of keys to make one key
+    if not isinstance(key, basestring):
+      new_key = [str(key_item) for key_item in key]
+      key = '#'.join(new_key)
```

Comment:
This should trigger nesting rather than suffixing like this. Will likely need to do it recursively.